lib/Common/CommonPal.h

```diff
@@ -31,17 +31,26 @@ typedef wchar_t char16;
 #else // !_WIN32
 
 #define USING_PAL_STDLIB 1
+#define STRSAFE_INLINE   1
+
 #ifdef PAL_STDCPP_COMPAT
 #include <math.h>
 #include <time.h>
 #include <smmintrin.h>
 #include <xmmintrin.h>
 #endif
+
 #include "inc/pal.h"
 #include "inc/rt/palrt.h"
 #include "inc/rt/no_sal2.h"
 #include "inc/rt/oaidl.h"
 
+// In general, we don't need to force-inline any of Chakra's functions
+// on Linux- we'd rather let the compiler figure this out for us
+// If, in the future, perf investigations reveal that force inlining
+// would help, we can address on a case-by-case basis.
+#define __forceinline inline
```

Comment:
Changed. I keep going back and forth on this, I wanted it to be inline initially so that we switch it to always_inline only after we've understood the perf. Switched to your model for now but might change my mind again in the future 😄